src/Arcus.Messaging.Tests.Integration/MessagePump/Fixture/MockCircuitBreakerEventHandler.cs

```diff
@@ -41,7 +47,7 @@ private static void VerifyCorrectTransition(
         {
             if (oldState.IsClosed)
             {
-                Assert.True(newState.IsHalfOpen, $"when the message pump comes from a closed state, the next state should always be half-open, but got: {newState}");
+                Assert.True(newState.IsOpen, $"when the message pump comes from a closed state, the next state should always be open, but got: {newState}");
```

Comment:
This was wrongly asserted but was not previously found since we did added 'fire-forget' to the message pump, so the test was always 'too fast' before the states were stored/verified here.

Changed it so that we first wait for the required state changes (3 in total) before we assert it.